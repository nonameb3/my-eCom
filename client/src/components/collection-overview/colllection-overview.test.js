import React from "react";
import { shallow } from "enzyme";

import CollectionOverview from "./collection-overview.component";
import CollectionPreview from "../../components/collection-preview/collection-preview.component";

let wrapper;
const mockCollection = [
  {
    title: "hats",
    imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
    id: 1,
    linkUrl: "shop/hats"
  },
  {
    title: "jackets",
    imageUrl: "https://i.ibb.co/px2tCc3/jackets.png",
    id: 2,
    linkUrl: "shop/jackets"
  }
];

jest.mock("react-redux", () => ({
  useSelector: jest.fn(() => ({ collections: mockCollection }))
}));

describe("collection-overview component", () => {
  beforeEach(() => {
    wrapper = shallow(<CollectionOverview />);
  });

  it("should render collection-overview component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render collection-preview as expect", () => {
    expect(wrapper.find(CollectionPreview).length).toBe(mockCollection.length);
  });
});
