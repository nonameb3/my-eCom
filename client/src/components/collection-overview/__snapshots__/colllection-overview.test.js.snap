// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection-overview component should render collection-overview component 1`] = `ShallowWrapper {}`;
